crates/swc_ecma_transforms_base/src/hygiene/usage_analyzer.rs

```diff
@@ -31,6 +31,7 @@ pub(super) struct ScopeData {
 pub(super) struct CurScope<'a> {
     pub parent: Option<&'a CurScope<'a>>,
     pub data: ScopeData,
+    pub depth: u16,
```

Comment:
What would probably be better is to instead provide `top_level_mark` to `hygiene(top_level_mark)` then that could be checked instead.

Reply:
Yeah, it's a big change and depth check is enough for now.

Reply:
I think `resolver` should distinguish top level bindings and unresolved references, so I filed https://github.com/swc-project/swc/issues/2956
Maybe in future `hygiene` pass can be updated to aware of it, but it's not a scope of this PR.

Reply:
Yes, that sounds better! Then you wouldn't need to pass around a top_level_mark, which can be disruptive.